plugins/modules/ipahbacrule.py

```diff
@@ -469,21 +469,36 @@ def main():
             elif state == "enabled":
                 if res_find is None:
                     ansible_module.fail_json(msg="No hbacrule '%s'" % name)
+                # FreeIPA 4.9.10+ and 4.10 use proper mapping for
+                # boolean values, so we need to convert it to str
+                # for comparison.
+                # See: https://github.com/freeipa/freeipa/pull/6294
+                is_enabled = (
+                    str(res_find["ipaenabledflag"][0]).upper() == "TRUE"
+                )
                 # hbacrule_enable is not failing on an enabled hbacrule
                 # Therefore it is needed to have a look at the ipaenabledflag
                 # in res_find.
-                if "ipaenabledflag" not in res_find or \
-                   res_find["ipaenabledflag"][0] != "TRUE":
+                # FreeIPA 4.9.10+ and 4.10 use proper mapping for boolean
+                # values, so only searching for "TRUE" does not work anymore.
+                # See: https://github.com/freeipa/freeipa/pull/6294
+                if "ipaenabledflag" not in res_find or not is_enabled:
```

Comment:
Do we need `if "ipaenabledflag" not in res_find` as `res_find["ipaenabledflag"]` was used before?

Reply:
More like this below.

Reply:
Fixed.